### DISCUSSION TITLE
Future Asset Libraries - What should they be?

### DISCUSSION THREAD

**Discussion by Draise14** [Bforartists/Bforartists#2296] [Ideas]
I have been taking a look at the asset management with the Asset Browser, still not mature, but it should be hopefully later this year. Meaning we can "be for artists" even more. 

I have also started making my own asset library, planning to contribute to this here in the future. There are many CC0 PBR textures and new node groups we could make, and also work many new things similar to Quixel or Substance - or Zbrush - or even something minimal like trueSpace. 

But I think what would make Bforaritsts really stand out is... it has libraries and presets to make life easier to work and pick up and use. This is what makes professional software more artist friendly and stand out compared to Blender. Presets and assets. 

### What libraries should be included with Bforartists? 

Tool Libraries:
- Sculpt Brush presets
- Grease Pencil Brushes (similar to the Blender Cloud ones)
- Texture Brush Presets
- Vertex Paint Brush Presets
- Colour Palettes
- Gradient Palettes (node groups)
- Smart Mask Nodes (node groups)
- Compositor Functions (node groups)
- Geometry Nodes Functions (node groups)
- Material Functions (node groups)
- Grunge/Dirt/Wear Functions (node groups)

Technical Examples:
- Modifier Stack examples
- Particle Examples
- Geometry Nodes examples
- Animation Examples
- Motion Tracking Examples
- Compositor Examples
- Video Editing Examples
- Animatic Examples
- Concept Art Examples
- Hard Surface Texture Examples
- Character Modeling Examples

Material Packs
_we already have a few libraries with VX LIbrary addon, it now would just need migration to the Asset Manager_
- CC0 PBR texture packs? (like re-create cc0textures.com materials)
- More procedural material examples
- CC0 Decal packs?

Models/Asset Packs?
- Rigged Characters
- Base Rig Characters Examples
- Foilage Packs
- Rock Packs
- Urban Packs
- Scifi Packs
- Midieval Packs
- NPR packs
- Low Poly packs
- Sculpt Base Mesh packs
- Arquitectural packs
- Industrial packs

Everything must be CC0 if by origin, or be set as CC0 for later use. 

I know the list seems huge, but over time small sections can be added or updated piece by piece, keeping Bforaritsts fresh and powerful. 

## Technicalities
To include assets to the build, it will make the file a little larger, or a lot larger if we include 3D assets (objects and PBR materials). 

Should we have these packs as blend files with marked assets as a separate download, or include them into the installation? 

Maybe a hybrid? Include materials and brushes and node groups by default, but the extra PBR material and models available on the website? 

Just want to plan this out, as I would love to dedicate some time to this, because this I can do - more than just coding and documentation. I just want to know of logistics, how to approach it, and how it can fit into the Bforaritsts vision. 

Do we want example files? 3D Assets? Tool libraries? Or just material libraries?

**Comment by ReinerBforartists** [comment]
Evruthing !!!! O_O

**Reply by Draise14** [reply]
Alright. I am churning the old brain. 

Biggest issue right now is node groups right now. Shader node groups, composite node groups, GN node groups and more all are lumped together in the datablocks system, so I will probably need a naming convention system. 

I'm also wondering, how to distribute, like.. a blend pack to add to the asset library? Extra Downloads from website? A stronger default? 

I am still mapping it out. I will do an asset proposal over time.

**Comment by ReinerBforartists** [comment]
Sounds great :)

We should definitely care about the download size though. Shaders and procedural materials doesn't eat up this much space. But textures do. Also, there is no need to have a full library right from the start. I think more of two or three examples per category. And add more over the time :)

**Comment by Draise14** [comment]
Blender has opened up their ideas to also create asset libraries. They are thinking to manage them as separate files I guess. 
[https://developer.blender.org/T55239](https://developer.blender.org/T55239)
[https://wiki.blender.org/wiki/Reference/Release_Notes/Blender_Asset_Bundle](https://wiki.blender.org/wiki/Reference/Release_Notes/Blender_Asset_Bundle)
[https://developer.blender.org/diffusion/BL/browse/trunk/lib/assets/
](https://developer.blender.org/diffusion/BL/browse/trunk/lib/assets/
)
To circumnavigate bandwidth, files could be hosted on other services.

**Comment by ReinerBforartists** [comment]
This is great. This saves us quite a bit of work then. Let's see what gaps they leave for us to fill :)

**Comment by Draise14** [comment]
Yes, I think this is good. I do think we can always additionally release new "packs" to updates to make Bforartists relevant each release - we should still cator to the artist, and sometimes an artists just wants to create art and not create the assets all the time. But yeah, bandwidth.
